packages/core/data-transfer/src/file/providers/destination/index.ts

```diff
@@ -252,13 +252,22 @@ class LocalFileDestinationProvider implements IDestinationProvider {
         // always write tar files with posix paths so we have a standard format for paths regardless of system
         const entryPath = path.posix.join('assets', 'uploads', data.filename);
 
+        const entryMetadataPath = path.posix.join('assets', 'metadata', `${data.filename}.json`);
+        archiveStream.entry(
+          {
+            name: entryMetadataPath,
+            size: JSON.stringify(data.metadata).length,
```

Comment:
JSON.stringify is quite slow, you could improve that by only parsing once & reusing the result here & L261